src/middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+type Environment = 'production' | 'development' | 'other';
+
+const domainsToUpgrade = ['spacebar.build', 'www.spacebar.build'];
+
+const upgradeToHTTPSMiddleware = (req: NextRequest) => {
+  const currentEnv = process.env.NODE_ENV as Environment;
+
+  // Heroku's load balancer forwards all requests via HTTP,
+  // so we check the x-forwarded-proto header to determine if the original request was made over HTTPS:
+  // https://stackoverflow.com/questions/66458059/how-do-i-force-or-redirect-my-next-js-website-to-use-https
+  if (
+    currentEnv === 'production' &&
+    req.headers.get('x-forwarded-proto') !== 'https' &&
+    domainsToUpgrade.includes(req.headers.get('host') ?? '')
+  ) {
+    return NextResponse.redirect(
+      `https://${req.headers.get('host')}${req.nextUrl.pathname}`,
+      301,
+    );
+  }
```

Comment:
can we explicitly check that the host is not nil (it shouldn't be but still) instead of falling back to empty string?
```suggestion
  const currentHost = req.headers.get('host');
  if (
    currentEnv === 'production' &&
    req.headers.get('x-forwarded-proto') !== 'https' &&
    currentHost &&
    domainsToUpgrade.includes(currentHost)
  ) {
    return NextResponse.redirect(
      `https://${currentHost}${req.nextUrl.pathname}`,
      301,
    );
  }
```

Reply:
Done!